sdk/core/Azure.Core/src/TokenCredential.cs

```diff
@@ -54,6 +55,41 @@ public override AuthenticationToken GetToken(GetTokenOptions properties, Cancell
             GetToken(TokenRequestContext.FromGetTokenOptions(properties), cancellationToken).ToAuthenticationToken();
 
         /// <inheritdoc />
-        public override GetTokenOptions? CreateTokenOptions(IReadOnlyDictionary<string, object> properties) => null;
+        public override GetTokenOptions? CreateTokenOptions(IReadOnlyDictionary<string, object> properties)
+        {
+            // Check if scopes are present and in a valid format
+            if (properties.TryGetValue(GetTokenOptions.ScopesPropertyName, out var scopesValue))
+            {
+                // Try to convert scopes to ReadOnlyMemory<string>
+                ReadOnlyMemory<string> scopes = default;
+
+                if (scopesValue is ReadOnlyMemory<string> readOnlyMemoryScopes)
+                {
+                    scopes = readOnlyMemoryScopes;
+                }
+                else if (scopesValue is string[] stringArrayScopes)
+                {
+                    scopes = new ReadOnlyMemory<string>(stringArrayScopes);
+                }
```

Comment:
why do we allow both ROM and string[]? If we required ROM, we  would not need all this logic in this rouitine

Reply:
having two options does not communicate this either. It's an advanced API. I would just require ROM and make the routine simpler/faster.